src/middleware/auth_middleware.go

```diff
@@ -0,0 +1,48 @@
+package middleware
+
+import (
+	"github.com/gin-gonic/gin"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"net/http"
+	"os"
+	"strings"
+)
+
+func TokenAuthMiddleware() gin.HandlerFunc {
+	return func(c *gin.Context) {
+		token := c.GetHeader("Authorization")
+		if token == "" {
+			c.JSON(http.StatusUnauthorized, gin.H{"error": "Authorization token not provided"})
+			c.Abort()
+			return
+		}
+
+		tokenParts := strings.Split(token, " ")
+		if len(tokenParts) != 2 || tokenParts[0] != "Bearer" {
+			c.JSON(http.StatusUnauthorized, gin.H{"error": "Invalid authentication token"})
+			c.Abort()
+			return
+		}
+
+		token = tokenParts[1]
```

Comment:
Please add a new function to extract the token from the authorization header. This one should work:
```
func getBearerToken(c *gin.Context) (string, error) {
    authHeader := c.GetHeader("Authorization")
    if authHeader == "" {
        return "", fmt.Errorf("Authorization header not found")
    }

    parts := strings.Split(authHeader, " ")
    if len(parts) != 2 || strings.ToLower(parts[0]) != "Bearer" {
        return "", fmt.Errorf("Invalid Authorization header format")
    }

    return parts[1], nil
}
```